pkg/git/service.go

```diff
@@ -188,6 +191,131 @@ func (s *Service) SetGitConfig(userData *gitprovider.GitUser) error {
 	return nil
 }
 
+func formatCommit(commit *object.Commit, branchName string) project.CommitInfo {
+	return project.CommitInfo{
+		Hash:    commit.Hash.String(),
+		Author:  commit.Author.Name,
+		Message: strings.TrimSpace(commit.Message),
+		Date:    commit.Author.When.Format("2006-01-02 15:04:05"),
+		Branch:  branchName,
+	}
+}
+
+func (s *Service) GetUnpushedCommitsInfo() ([]project.CommitInfo, error) {
+	r, err := git.PlainOpen(s.ProjectDir)
+	if err != nil {
+		log.Debugf("Error opening repository: %v", err)
+		return nil, nil
+	}
+
+	ref, err := r.Head()
+	if err != nil {
+		log.Debugf("Error getting HEAD reference: %v", err)
+		return nil, nil
+	}
+	branchName := ref.Name().Short()
+
+	mainRef, err := r.Reference(plumbing.ReferenceName(fmt.Sprintf("refs/heads/%s", branchName)), true)
+	if err != nil {
+		log.Debugf("Local branch '%s' does not exist: %v", branchName, err)
+		return nil, nil
+	}
+
+	originMainRef, err := r.Reference(plumbing.ReferenceName(fmt.Sprintf("refs/remotes/origin/%s", branchName)), true)
+	if err != nil {
+		if err == plumbing.ErrReferenceNotFound {
+			defaultBranch, err := getDefaultBranch(s.ProjectDir)
```

Comment:
Why do we default back to the default branch here? Why don't we just print out that the branch is not pushed to remote?

Reply:
@Tpuljak We default to the default branch to accurately count unpushed commits because Git-Go does not provide a direct way to determine unpushed commits for a local branch that hasn't been pushed to the remote. By comparing with the default branch, we can effectively identify and list the unpushed commits.

Reply:
> BTW are you here referring to showing the message(branch not published) on the dtn info instead of number of unpushed commits.

Exactly 😄 

That's the simplest solution that paints the whole picture actually. Also, the problem with your approach is that the branch might be created from another branch, not the default one. Then the count would be completely wrong.

Reply:
OK got it . WIP

Reply:
Done
.